### PR TITLE
[all][website] Apply font-smoothing to all components

### LINKED ISSUE #552: Font smoothing is not set to antialiased [Design]
### Expected behavior

On OS X, fonts weights, such as 600 semi-bold should render smoother and more akin to the design of the typeface than they appear currently. 

### Current behavior

 Font smoothing is not currently applied. 

### Possible solution

Recommend applying `-webkit-font-smooth: antialiased;` and `-moz-osx-font-smoothing: auto`

### PR DESCRIPTION
**Pull request by kylegach**
### Description

- Apply font-smoothing, via `componentStyleReset`, to all components
- Various small tweaks to site styles that were necessary afterwards
- Update snapshots

### Motivation and context

Closes #552 

### Screenshots, videos, or demo, if appropriate

https://552-font-smoothing--mineral-ui.netlify.com/

### How to test

1. `cd mineral-ui && npm start`
2. Browse around, make sure things look good

### Types of changes

- New feature (non-breaking change which adds functionality)
- Other (provide details below)

Other: website tweaks

### Checklist

* [x] Renders and functions properly in [all supported browsers](https://github.com/mineral-ui/mineral-ui#browser-support)
* [x] Automated tests written and passing [n/a]
* [x] [Accessibility](http://webaim.org/intro) and [inclusivity](https://24ways.org/2016/what-the-heck-is-inclusive-design/) considered
* [x] Rendering performance (initial load time & 60fps) and [perceived performance](http://blog.teamtreehouse.com/perceived-performance) considered [n/a]
* [x] Documentation created or updated [n/a]
* [x] Tested in [sandbox](https://github.com/facebookincubator/create-react-app#getting-started) if new component or breaking change [n/a]

### How does this PR make you feel?

Michael Jackson being smooth

### CODE REVIEW

File: src/website/app/pages/Home/index.js [hunk 1]
```diff
@@ -115,7 +114,7 @@ const gettingStartedTheme = {
   SiteLink_color_hover: siteColors.yellow_hover
 };
 const buttonTheme = {
-  Button_fontWeight: mineralTheme.fontWeight_regular,
+  Button_fontWeight: 500,
```
Comment (coldpour) [inline src/website/app/pages/Home/index.js]: shouldn't this be using the new variable? it seems strange that this file has a bunch of static themes in it where the rest of our stuff uses functions

Comment (VictoriaVasys) [reply src/website/app/pages/Home/index.js]: Good points. We should definitely use the new variable. I don't know enough about the theming to identify what needs to be static here, but @kylegach if you want to help me identify what can be improved, I'm happy to do that.

**Comment by VictoriaVasys** [comment]
@coldpour the finer text is a result of the font smoothing. I'm curious as to why the svg looks darker though.

**Comment by coldpour** [comment]
the svg looks darker because it changes from light to dark and back again over a period of a minute or two. nothing to worry about.

**Pull request merged**